File: android/src/main/java/me/haymob/coffeeshop/android/components/CatalogProductItem.kt
package me.haymob.coffeeshop.android.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import me.haymob.coffeeshop.android.Gallery
import me.haymob.coffeeshop.android.R
import me.haymob.coffeeshop.android.extensions.string
import me.haymob.coffeeshop.entities.Product

@Composable
fun CatalogProductItem(product: Product, onClick: () -> Unit, inc: () -> Unit, dec: () -> Unit) {
    Box(
        modifier = Modifier
            .fillMaxSize()
            .padding(10.dp)
            .clip(RoundedCornerShape(20.dp))
    ) {
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(2.dp),
            modifier = Modifier
                .clickable(onClick = onClick)
                .background(Color.White)
        ) {
            Column(
                modifier = Modifier
                    .padding(8.dp)
                    .clip(RoundedCornerShape(20.dp))
                    .background(Color.Gallery)
            ) {
                ProductImage(
                    product.thumbnail,
                    Modifier
                        .padding(12.dp)
                        .fillMaxWidth()
                        .aspectRatio(1.0f)
                )
            }
            Text(
                product.name,
                textAlign = TextAlign.Center,
                fontSize = 17.sp
            )
            Text(
                product.price.string,
                fontWeight = FontWeight.Bold,
                textAlign = TextAlign.Center,
                fontSize = 17.sp
            )
            if (product.qty == 0) {
                ActionButton(R.drawable.ic_plus, 60.dp, inc)
            } else {
                ActionButtons(product.qty, inc = inc, dec = dec)
            }
        }
        if (product.isLoading) Loader(Modifier.matchParentSize())
    }
}

File: ios/ios/Views/CatalogProductItem.swift
import SwiftUI
import core

func CatalogProductItem(product: Product, inc: @escaping () -> Void, dec:  @escaping () -> Void) -> some View {
    ZStack {
        VStack {
            VStack {
                ProductImage(image: product.thumbnail)
            }
            .padding()
            .frame(maxWidth: .greatestFiniteMagnitude, maxHeight: .greatestFiniteMagnitude)
            .aspectRatio(1, contentMode: .fill)
            .background(Color.gallery)
            .cornerRadius(20)
            Text(product.name)
                .font(.appRegular)
            Text("\(product.price)")
                .font(.appSemibold)
            if product.qty == 0 {
                ActionButton(image: "plus", width: 60, action: inc)
            } else {
                ActionButtons(qty: product.qty, inc: inc, dec: dec)
            }
        }
        .padding(6)
        .background(.white)
        .cornerRadius(20)
        if product.isLoading {
            ProductLoader()
        }
    }
}
